client/app.js

```diff
@@ -30,6 +30,7 @@ tie.constant('CODE_EXECUTION_TIMEOUT_SECONDS', 3);
 
 // Supported languages.
 tie.constant('LANGUAGE_PYTHON', 'python');
+tie.constant('ALL_SUPPORTED_LANGUAGES', ['python']);
```

Comment:
Nope -- I felt sad about this one too. Need to hardcode, otherwise it becomes too complicated for a constant. Can probably find a better way when we actually support more languages, and all the constants are in the same location anyway so I was comfortable punting it (since anyone adding a new language is going to see this).